RAGchain/benchmark/base.py

```diff
@@ -99,7 +101,10 @@ def _calculate_metrics(self,
                 'contexts': result_df['passage_contents'].tolist()
             }
             if retrieval_gt is not None:
-                dataset_dict['ground_truths'] = self.__fetch_contents(retrieval_gt)
+                result_df['retrieval_gt_contents'] = result_df.apply(
+                    lambda row: [passage.content for passage in self.dummy_retrieval.fetch_data(row['retrieval_gt'])],
+                    axis=1)
+                dataset_dict['ground_truths'] = result_df['retrieval_gt_contents'].tolist()
 
             ragas_result = evaluate(
                 Dataset.from_dict(dataset_dict),
```

Comment:
In line 96 to 102 at benchmark/base.py, if passages that is not combined retrieval_gt are not ingested, `self.__fetch_contents` can't find retrieval_gt contents. So `dataset_dict` might be an empty list. It occur value error like below.



![image](https://github.com/NomaDamas/RAGchain/assets/60510718/580467b8-dded-42aa-a10f-6e114883bed2)